app/serializers/chat_message_serializer.rb

```diff
@@ -103,4 +104,13 @@ def user_flag_status
   def include_user_flag_status?
     user_flag_status.present?
   end
+
+  def hidden
+    return false if scope.anonymous?
+    ignored_user_ids.include?(object.user_id)
+  end
+
+  def ignored_user_ids
+    @ignored_user_ids ||= scope.user.ignored_user_records.pluck(:ignored_user_id)
```

Comment:
is this called for each rendered message?

Reply:
Yes, unless the user is anonymous, however it's memoized so shouldn't cause an N1 (and the user may already have some of this preloaded in core, will have to double check)

Reply:
Yes please let's double check this, thanks.

Reply:
We actually don't preload anything for the current user in the default provider:

https://github.com/discourse/discourse/blob/a24bcceb19cb7272f1725d3fe5e2dd6be2bc0128/lib/auth/default_current_user_provider.rb#L156-L156

The current user serializer also loads this data:

https://github.com/discourse/discourse/blob/a24bcceb19cb7272f1725d3fe5e2dd6be2bc0128/app/serializers/current_user_serializer.rb#L237-L239

So maybe it does make sense to only filter on client side? Otherwise we end up with both this and current user serializer querying this data :thinking:

Reply:
or maybe core should be changed to preload this ?

Reply:
Yeah not sure of the implications of preloading more things on the User in core via default user provider, but maybe it makes sense to load these and maybe user options? Just hope that it doesn't make a performance hit, will try it out

Reply:
Actually I am not sure it will make a difference to preload in core seeing as how this plucks IDs and core plucks usernames. I think I will just remove this and rely on the frontend filtering in `_prepareSingleMessage`